pkg/update/update.go

```diff
@@ -56,21 +56,37 @@ func makeUpdateImagesFilter(originalRepo, replacement string) kio.Filter {
 	})
 
 	return kio.FilterFunc(func(objs []*yaml.RNode) ([]*yaml.RNode, error) {
+		tees := []yaml.Filter{
+			yaml.Tee(
+				yaml.Lookup("initContainers"),
+				replaceImageInEachContainer,
+			),
+			yaml.Tee(
+				yaml.Lookup("containers"),
+				replaceImageInEachContainer,
+			),
+		}
+
 		for _, obj := range objs {
-			if err := obj.PipeE(
-				yaml.Lookup("spec", "template", "spec"),
-				yaml.Tee(
-					yaml.Lookup("initContainers"),
-					replaceImageInEachContainer,
-				),
-				yaml.Tee(
-					yaml.Lookup("containers"),
-					replaceImageInEachContainer,
-				),
-			); err != nil {
-				return nil, err
+			switch kind(obj) {
```

Comment:
As the case bodies differ only on the path given to `yaml.Lookup`, could just that be varied inside the switch and the pipe be run with that value afterward? Or are you anticipating other varieties of pipe.

Reply:
Good point, I've simplified this a bit.